### ISSUE TITLE
asset+tapgarden+vm: Update group key signing to produce a witness over a minting virtual TX

### ISSUE THREAD

**Issue by jharveyb** [lightninglabs/taproot-assets#466]
Sub-issue of #394, depends on PRs #448 and #458 or similar.

As part of the asset and proof spec changes, asset membership in a group is now authorized by a generic witness stack and not a signature over the asset `Genesis` data. This witness must be validated over a TX, which will now be the virtual TX that represents an asset minting, as specified here https://github.com/Roasbeef/bips/pull/33 .

This requires new logic to construct these minting virtual TXs, adjustments of VM (and possibly other) logic to allow an asset witness on minting if and only if the asset has a group, and further extensions to the VM to actually validate this new witness if the asset is being minted and has a group.

I *think* these changes should be limited mostly to the VM but I'm not sure. Fixing other subsystems may be deferred to a future PR.

**Requirements**

- [ ] New logic to construct virtual TXs for asset minting
- [ ] Modify group key signing logic to sign over the minting virtual TX
- [ ] Modify minting to include the group key witness in the `Asset` PrevWitness field
- [ ] New logic to verify that a virtual TX is for asset minting
- [ ] Modify VM asset minting exception to verify the witness for asset group mints
- [ ] New VM unit tests for asset group witnesses that use the script path

Pessimistically sizing this as Medium due to uncertainty around the exact format of the virtual TX and the VM changes needed to get this working.

**Comment by dstadulis** [comment]
The virtual tx construction is somewhat unclear in the BIP -- and may not be entirely what we want to do.  JHB to provide more context

**Comment by jharveyb** [comment]
Linking to BIP comment about the exact structure of the virtual TX and some underspecification there:

https://github.com/Roasbeef/bips/pull/33#issuecomment-1690271321
